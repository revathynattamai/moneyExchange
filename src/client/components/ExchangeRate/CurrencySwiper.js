import React from 'react';
import SwipeableViews from 'react-swipeable-views';

const CurrencySwiper = ({ currencies, onChange, onChangeIndex, ...otherProps }) => {

  return <div>
    <SwipeableViews onChangeIndex={onChangeIndex} style={{ height: 50, border: 'solid 1px black' }}>
      {Object.keys(currencies).map((currency, i) => <div key={i}>
        {currency}
        <input type="text" onChange={e => onChange(e, currency)} {...otherProps} />
        {currencies[currency]}
      </div>)}
    </SwipeableViews>
  </div>
};

export default CurrencySwiper;