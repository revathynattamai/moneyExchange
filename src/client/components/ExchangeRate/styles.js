import { textAlign } from "@material-ui/system";

export default {
  main: {
    padding: 10,
    textAlign: "center",
    fontFamily: "verdana",
    maxWidth: 300,
    fontSize: 12,
    border: "solid 1px #ccc",
    background: "white"
  },
  title: {
    color: "black", textAlign: "center", fontSize: 30, padding: 10,
    margin: 10
  },
  paper: {
    height: 50, 
    margin: 10,
    border: "solid 1px #ccc"
  }
};