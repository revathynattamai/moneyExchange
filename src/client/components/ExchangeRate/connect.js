import { connect } from 'react-redux';

import { getCurrencies, exchange } from '../../data/currencies/currenciesActions';
import { getExchangeRates } from '../../data/exchangeRates/exchangeRatesActions';
// import { setValues, exchangePockets } from '../../data/app/appActions';
import ExchangeRate from './ExchangeRate';

const mapStateToProps = state => state;
const mapDispatchToProps = dispatch => ({
  getCurrencies: data => dispatch(getCurrencies(data)),
  getExchangeRates: data => dispatch(getExchangeRates(data)),
  exchange: data => dispatch(exchange(data)),
  // setValues: (val, pocket, pocketTo, direction) => dispatch(setValues(val, pocket, pocketTo, direction)),
  // exchangePockets: () => dispatch(exchangePockets()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExchangeRate);