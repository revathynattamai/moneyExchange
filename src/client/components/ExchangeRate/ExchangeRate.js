import React, { Component } from 'react';
import { connect } from 'react-redux';

import SwipeableViews from 'react-swipeable-views';
import TextField from '@material-ui/core/TextField';
import classNames from 'classnames';
import { Typography } from '@material-ui/core';
import Fab from '@material-ui/core/Fab';
import { withStyles } from '@material-ui/core/styles';
import styles from './styles';
import MobileStepper from '@material-ui/core/MobileStepper';
import Button from '@material-ui/core/Button';
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import { convert } from '../../data/inputs/inputActions';
import withErrorBoundary from './errorBoundary';

import { formatMoney } from '../../utils/helpers';

class ExchangeRate extends Component {
  constructor(props) {
    super(props);
    this.state = {
      indexTop: 0,
      indexBottom: 1,
    }
  }

  componentDidMount = () => {
    const { getCurrencies, getExchangeRates } = this.props;
    getCurrencies();
    // getExchangeRates();
    // window.setInterval(getExchangeRates, 10000);
  }

  render() {
    const { currencies } = this.props;
    if(!currencies) return null;
    return (<div>
      {Object.keys(currencies).map((a, i) => <SwipeableViews key={i}> {a} </SwipeableViews>)}
      </div>);
  }
}

  export default withErrorBoundary(ExchangeRate);