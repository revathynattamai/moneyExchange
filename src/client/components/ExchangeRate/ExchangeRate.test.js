import React from 'react';
import ExchangeRate from './ExchangeRate';
import renderer from 'react-test-renderer';

describe('ExhcangeRate', () => {
  test('renders nothing if exchangeRate and currencies are null', () => {
    const props = {
      exchangeRate: null,
      currencies: null,
      getCurrencies: jest.fn(),
      getExchangeRates: jest.fn(),
    }
    const tree = renderer
      .create(<ExchangeRate {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders nothing if exchangeRate and currencies are null', () => {
    const props = {
      exchangeRate: {
        usd: 1,
        gbp: 2,
      },
      currencies: {
        usd: 10,
        gbp: 10,
      },
      getCurrencies: jest.fn(),
      getExchangeRates: jest.fn(),
    }
    const tree = renderer
      .create(<ExchangeRate {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
