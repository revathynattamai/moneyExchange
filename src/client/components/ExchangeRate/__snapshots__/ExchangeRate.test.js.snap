// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExhcangeRate renders nothing if exchangeRate and currencies are null 1`] = `null`;

exports[`ExhcangeRate renders nothing if exchangeRate and currencies are null 2`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthSm"
  style={
    Object {
      "backgroundColor": "#cfe8fc",
      "height": "100vh",
    }
  }
>
  <p
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "padding": 10,
        "textAlign": "center",
        "textShadow": "1px 1px 2px black, 0 0 25px blue, 0 0 5px darkblue",
      }
    }
  >
    Revolut
  </p>
  <div
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onScroll={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "height": 50,
        "margin": 10,
        "overflowX": "hidden",
        "width": "100%",
      }
    }
  >
    <div
      className="react-swipeable-view-container"
      style={
        Object {
          "WebkitFlexDirection": "row",
          "WebkitTransition": "all 0s ease 0s",
          "direction": "ltr",
          "display": "flex",
          "flexDirection": "row",
          "height": null,
          "transition": "all 0s ease 0s",
          "willChange": "transform",
        }
      }
    >
      <div
        aria-hidden={false}
        data-swipeable="true"
        style={
          Object {
            "WebkitFlexShrink": 0,
            "flexShrink": 0,
            "overflow": "auto",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <span
            style={
              Object {
                "fontSize": 20,
                "margin": 2,
              }
            }
          >
            USD
          </span>
          <input
            className="input"
            onChange={[Function]}
            placeholder="0.00"
            size="8"
            type="text"
            value=""
          />
          <span
            style={
              Object {
                "fontSize": 20,
                "margin": 2,
              }
            }
          >
            10.00
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onScroll={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "height": 50,
        "margin": 10,
        "overflowX": "hidden",
        "width": "100%",
      }
    }
  >
    <div
      className="react-swipeable-view-container"
      style={
        Object {
          "WebkitFlexDirection": "row",
          "WebkitTransition": "all 0s ease 0s",
          "direction": "ltr",
          "display": "flex",
          "flexDirection": "row",
          "height": null,
          "transition": "all 0s ease 0s",
          "willChange": "transform",
        }
      }
    >
      <div
        aria-hidden={false}
        data-swipeable="true"
        style={
          Object {
            "WebkitFlexShrink": 0,
            "flexShrink": 0,
            "overflow": "auto",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <span
            style={
              Object {
                "fontSize": 20,
                "margin": 2,
              }
            }
          >
            USD
          </span>
          <input
            className="input"
            disabled={true}
            onChange={[Function]}
            placeholder={0}
            size="8"
            type="text"
            value="0.00"
          />
          <span
            style={
              Object {
                "fontSize": 20,
                "margin": 2,
              }
            }
          >
            10.00
          </span>
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Exchange
    </span>
  </button>
  <p>
    Swipe to see other currencies
  </p>
</div>
`;
